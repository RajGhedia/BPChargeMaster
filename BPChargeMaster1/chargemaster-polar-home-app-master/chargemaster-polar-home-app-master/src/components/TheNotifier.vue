<template>
    <v-snackbar
        class="ma-2"
        v-model="notification"
        top
        auto-height
        :color="alertType"
        :timeout="0"
    >
        <span class="text-xs-center">{{ message }}</span>
    </v-snackbar>
</template>
<script>
import EventBus from '@/event-bus'
export default {
    data: () => {
        return {
            message: '',
            alertType: 'success',
            notification: false
        }
    },
    methods: {
        showNotification(payload) {
            if (typeof payload.timeout === 'undefined') {
                payload.timeout = 5000
            }

            this.alertType = payload.type || 'primary'
            this.message = payload.message
            this.timeout = payload.timeout
            this.notification = true
            // Have to do it like this because the component has an issue
            // with toggling timeout between indefinite and a duration
            if (payload.timeout > 0) {
                setTimeout(() => {
                    this.notification = false
                }, payload.timeout)
            }
        }
    },
    created() {
        EventBus.$on('notify', this.showNotification)
    },
    beforeDestroy() {
        EventBus.$off('notify', this.showNotification)
    }
}
</script>
<style>
.v-snack__content span {
    width: 100%;
}

.v-snack {
    z-index: 4000 !important;
}
</style>
