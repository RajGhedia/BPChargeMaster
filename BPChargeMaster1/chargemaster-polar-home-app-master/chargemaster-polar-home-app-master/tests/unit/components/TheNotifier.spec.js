import TheNotifier from '@/components/TheNotifier.vue'
import Vue from 'vue'
import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import EventBus from '@/event-bus'

const localVue = createLocalVue()
Vue.use(Vuetify)

describe('TheNotifier.vue', () => {
    it('renders empty by default', () => {
        const wrapper = getWrapper()

        expect(wrapper.element).toMatchSnapshot()
    })

    it('listens for the notify event', () => {
        const wrapper = getWrapper()

        EventBus.$emit('notify', {
            type: 'error',
            message: 'example error message',
            timeout: 10000
        })

        expect(wrapper.vm.message).toBe('example error message')
        expect(wrapper.vm.alertType).toBe('error')
        expect(wrapper.vm.timeout).toBe(10000)
        expect(wrapper.vm.notification).toBe(true)
    })

    it('defaults to 5 second timeout if not supplied', () => {
        const wrapper = getWrapper()

        EventBus.$emit('notify', {
            type: 'warn',
            message: 'example warn message'
        })

        expect(wrapper.vm.message).toBe('example warn message')
        expect(wrapper.vm.alertType).toBe('warn')
        expect(wrapper.vm.timeout).toBe(5000)
        expect(wrapper.vm.notification).toBe(true)
    })

    it('defaults to primary color if no type supplied', () => {
        const wrapper = getWrapper()

        EventBus.$emit('notify', {
            message: 'example info message'
        })

        expect(wrapper.vm.message).toBe('example info message')
        expect(wrapper.vm.alertType).toBe('primary')
        expect(wrapper.vm.timeout).toBe(5000)
        expect(wrapper.vm.notification).toBe(true)
    })

    it('cleans up event listener when component is destroyed', () => {
        const wrapper = getWrapper()

        wrapper.setMethods({
            showNotification: jest.fn()
        })

        const showNotification = wrapper.vm.showNotification
        const cleanup = jest.fn()

        EventBus.$off = cleanup
        wrapper.destroy()

        expect(cleanup).toBeCalledWith('notify', showNotification)
        EventBus.$emit('notify', { message: 'any message' })
        expect(showNotification).not.toHaveBeenCalled()
    })
})

function getWrapper() {
    return mount(TheNotifier, {
        localVue
    })
}
