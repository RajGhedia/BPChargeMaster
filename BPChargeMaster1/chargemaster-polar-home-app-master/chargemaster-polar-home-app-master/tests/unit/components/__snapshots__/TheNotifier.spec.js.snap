// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNotifier.vue renders empty by default 1`] = `<!---->`;
